compiler/cpp/src/thrift/generate/t_generator.h

```diff
@@ -96,7 +97,47 @@ class t_generator {
     return escape_string(constval->get_string());
   }
 
+  /**
+   * Check if all identifiers are valid for the target language
+   */
+  virtual void validate_input() const;
+
 protected:
+  virtual std::set<std::string> lang_keywords() {
+    std::string keywords[] = { "BEGIN", "END", "__CLASS__", "__DIR__", "__FILE__", "__FUNCTION__",
```

Comment:
Would prefer to see code and data in the implementation file, but not a blocker.

Reply:
Done